library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/CoGroup.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.accumulation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * CoGroup Join Accumulation.
+ */
+public class CoGroup<T> extends Combine<T>
+{
+  public CoGroup()
+  {
+    //for kryo
+  }
+
+
+  @Override
+  public List<List<T>> getOutput(List<Set<T>> accumulatedValue)
```

Comment:
Any thoughts on efficiency? This will create a lot of GC work. Maybe we should benchmark this operator to see what limits throughput?